workbench-core/authorization/src/dynamicAuthorization/dynamoDBDynamicPermissionsPlugin.test.ts

```diff
@@ -0,0 +1,664 @@
+      mockDDB.on(ScanCommand).resolves({
+        Count: 0
+      });
+
+      mockDDB
+        .on(ScanCommand, {
+          TableName: ddbTableName,
+          IndexName: getResourceBySortKeyIndex
+        })
+        .rejects(mockResourceNotFoundException);
+      try {
+        await dynamoDBDynamicPermissionsPlugin.init({});
+        expect.hasAssertions();
+      } catch (err) {
+        expect(err).toBeInstanceOf(BadConfigurationError);
+        expect(err.message).toBe('DynamoDB is not configured');
+      }
+    });
+
+    test('Initialize with admin group and admin permissions', async () => {
+      mockDDB.on(ScanCommand).resolves({
+        Count: 0
+      });
+      mockDDB.on(UpdateItemCommand).resolves({});
+      const adminId = 'adminId';
+      const adminPermissions: IdentityPermission[] = [
+        {
+          effect: 'ALLOW',
+          action: 'CREATE',
+          subjectId: 'sampleId',
+          subjectType: 'sampleType',
+          identityId: adminId,
+          identityType: 'GROUP'
+        },
+        {
+          effect: 'ALLOW',
+          action: 'UPDATE',
+          subjectId: 'sampleId',
+          subjectType: 'sampleType',
+          identityId: adminId,
+          identityType: 'GROUP'
+        }
+      ];
+      const response = await dynamoDBDynamicPermissionsPlugin.init({
+        groupsToBeCreated: {
+          groupIds: [adminId]
+        },
+        identityPermissionsToBeCreated: adminPermissions
+      });
+      expect(response.success).toBe(true);
+    });
+  });
+
+  describe('DynamicPermissions Functions', () => {
+    const sampleGroupId = 'sampleGroup';
+    const sampleUserId = 'sampleUser';
+
+    const sampleAssignedPK = `assigned#USER#${sampleUserId}`;
+    const sampleAssignedSK = `assigned#GROUP#${sampleGroupId}`;
+
+    const sampleGroupIdentity = `ip#GROUP#${sampleGroupId}`;
+    const sampleAction: Action = 'CREATE';
+    const sampleEffect: Effect = 'ALLOW';
+    const sampleSubjectType = 'sampleSubjectType';
+    const sampleSubjectId = 'sampleSubjectId';
+    const sampleIpPK = `ip#${sampleSubjectType}#${sampleSubjectId}`;
+    const sampleIpSK = `ip##${sampleGroupId}`;
+    const sampleConditions = {};
+    const sampleFields: string[] = [];
+
+    const mockIdentityPermissionItem = {
+      pk: { S: sampleIpPK },
+      sk: { S: sampleIpSK },
+      action: { S: sampleAction },
+      effect: { S: sampleEffect },
+      subjectType: { S: sampleSubjectType },
+      subjectId: { S: sampleSubjectId },
+      identity: { S: sampleGroupIdentity },
+      conditions: { M: sampleConditions },
+      fields: { L: [] }
+    };
+
+    const mockIdentityPermission: IdentityPermission = {
+      action: sampleAction,
+      effect: sampleEffect,
+      subjectType: sampleSubjectType,
+      subjectId: sampleSubjectId,
+      identityId: sampleGroupId,
+      identityType: 'GROUP',
+      conditions: sampleConditions,
+      fields: sampleFields
+    };
```

Comment:
do we need `as const` on any of these fields?

Reply:
Some of these fields are already constants,

Reply:
converted mockIdentityPermission to `as const`